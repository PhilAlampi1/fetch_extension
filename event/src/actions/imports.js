export const setFormMappingData = (sfId, riId) => ({
    type: 'SET_FORM_MAPPING_DATA',
    formMappingStandardFieldId: sfId,
    formMappingRowIdentifierId: riId
})

export const createUpdateUserFormFieldMappingInDb = () => ({
    type: 'CREATE_UPDATE_USER_FORM_FIELD_MAPPING_IN_DB'
})

export const resetFormMappingFields = () => ({
    type: 'RESET_FORM_MAPPING_FIELDS',
    formMappingRowIdentifierId: null,
    formMappingStandardFieldId: null,
    enteredDefaultValue: null,
    enteredDefaultOverride: null,
    rightClickedFormElementType: null,
    rightClickedFormElementValue: null,
    rightClickedFormElementOptions: null,
    defaultValueConfirmed: false,
    defaultOverrideConfirmed: false,
    formFieldSelector: null,
    rightClickSelectionIsValid: false
})