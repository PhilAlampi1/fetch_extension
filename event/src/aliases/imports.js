import {
    status,
    json,
    serverPath,
    formatForXhr,
    findAndStoreImportFileSetups,
    fetchStubValues,
    findAndStoreUserForms,
    setupContextMenu
} from '../utilities/utilities'

export const updateExistingImportFileNameInDb = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        fetch(serverPath + 'updateimportfilesetupname/' +
            state.imports.selectedImportFileSetupId + '/' +
            state.auth.userToken + '/' +
            state.imports.selectedImportFileSetupName
        )
            .then(json)
            .then(result => {
                return dispatch({
                    type: 'CONFIRM_EXISTING_IMPORT_FILE_NAME',
                    importFileNameConfirmed: true
                })
            })
    }
}

export const updateImportFieldMappingsInDb = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        const importFieldMappings = state.imports.importFieldMappingDbUpdates
        const tok = state.auth.userToken
        const data = formatForXhr([{
            name: 'values',
            value: JSON.stringify(importFieldMappings)
        }, {
            name: 'userToken',
            value: tok
        }])
        const xhr = new XMLHttpRequest()
        xhr.onreadystatechange = () => {
            if (xhr.readyState == XMLHttpRequest.DONE) {
                const res = JSON.parse(xhr.responseText)
                if (res.success) {
                    return dispatch({
                        type: 'COMPLETE_UPDATE_IMPORT_FIELD_MAPPINGS_IN_DB',
                        importFieldMappingDbUpdates: []
                    })
                } else {
                    // TODO - ADD ERROR HANDLING TO REDUCERS AND THROW ERROR HERE
                    // UPDATE UI COMPONENT TO DISPLAY ERROR
                    console.log('DATABASE REQUEST ERROR', res)
                }
            }
        }
        xhr.open("POST", serverPath + 'importfieldmappings/update')
        xhr.setRequestHeader("content-type", "application/x-www-form-urlencoded");
        xhr.setRequestHeader("cache-control", "no-cache");
        xhr.send(data)
    }
}

export const createImportFileSetup = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        fetch(serverPath + 'createimportfilesetup/' +
            state.imports.selectedImportFileSetupName + '/' +
            state.auth.userId + '/' +
            state.auth.userToken
        )
            .then(json)
            .then(result => {
                // Insert all Import_Field_Mapping records into DB
                // Create Import_Field_Mapping array out of state
                let importFieldMappings = []
                const standardFieldsCopy = state.imports.standardFields
                standardFieldsCopy.map(sf => {
                    if (sf.importedFieldName) {
                        importFieldMappings.push({
                            userId: state.auth.userId,
                            importFileSetupId: result.data.importFileSetupId,
                            standardFieldId: sf.standardFieldId,
                            importedFieldName: sf.importedFieldName
                        })
                    }
                })
                const tok = state.auth.userToken
                const data = formatForXhr([{
                    name: 'values',
                    value: JSON.stringify(importFieldMappings)
                }, {
                    name: 'userToken',
                    value: tok
                }])
                const xhr = new XMLHttpRequest()
                xhr.onreadystatechange = () => {
                    if (xhr.readyState == XMLHttpRequest.DONE) {
                        const res = JSON.parse(xhr.responseText)
                        if (res.success) {
                            findAndStoreImportFileSetups(state.auth.userId, state.auth.userToken, dispatch)
                            return dispatch({
                                type: 'COMPLETE_CONFIRM_NEW_IMPORT_FILE_NAME',
                                selectedImportFileSetupId: result.data.importFileSetupId,
                                importFileNameConfirmed: true
                            })
                        } else {
                            // TODO - ADD ERROR HANDLING TO REDUCERS AND THROW ERROR HERE
                            // UPDATE UI COMPONENT TO DISPLAY ERROR
                            console.log('DATABASE REQUEST ERROR', res)
                        }
                    }
                }
                xhr.open("POST", serverPath + "importfilemappings/add")
                xhr.setRequestHeader("content-type", "application/x-www-form-urlencoded");
                xhr.setRequestHeader("cache-control", "no-cache");
                xhr.send(data)
            })
    }
}

export const createNewForm = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        fetch(serverPath + 'createform/' +
            state.imports.selectedFormName + '/' +
            state.imports.selectedFormDescription + '/' +
            state.auth.userToken
        )
            .then(json)
            .then(result => {
                return dispatch({
                    type: 'SET_SELECTED_FORM_ID',
                    selectedFormId: result.data.formId
                })
            })
    }
}

export const getAndStoreImportFieldMappings = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        fetch(serverPath + 'findimportfieldmappings/' +
            state.imports.selectedImportFileSetupId + '/' +
            state.auth.userToken
        )
            .then(json)
            .then(result => {
                // Instead of returing ImportFieldMappings, update standardFields for the user's selected Import File Setup
                // and replace the old standardFields with the new standardFields in the store               
                return dispatch({
                    type: 'STORE_IMPORT_FIELD_MAPPINGS',
                    standardFields: result.data
                })
            })
    }
}

export const fetchStubValuesFromAlias = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        const currentPage = state.imports.usersCurrentPage
        fetchStubValues()
            .then(() => {
                findAndStoreImportFileSetups(state.auth.userId, state.auth.userToken, dispatch)
                findAndStoreUserForms(state.auth.userToken, dispatch)
                if (currentPage !== 'main') {
                    return dispatch({
                        type: 'UPDATE_USERS_CURRENT_PAGE',
                        usersCurrentPage: 'main'
                    })
                }
            })
    }
}

export const updateFormInDb = () => {
    return (dispatch, getState) => {
        const state = getState() // had to use state here because alias wasn't picking up arguements passed in from popup
        fetch(serverPath + 'updateform/' +
            state.imports.selectedFormId + '/' +
            state.imports.selectedFormName + '/' +
            state.imports.selectedFormDescription + '/' +
            state.imports.selectedFormPublic + '/' +
            state.auth.userToken
        )
            .then(json)
            .then(result => {
                return dispatch({
                    type: 'CONFIRM_UPDATE_FORM',
                    formConfirmed: true
                })
            })
    }
}

export const activateContextMenus = () => {
    return () => setupContextMenu()
}

export const removeContextMenus = () => {
    return () => chrome.contextMenus.removeAll()
}

