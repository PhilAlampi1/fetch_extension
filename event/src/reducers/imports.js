const imports = {}

export default (state = imports, action) => {
  switch (action.type) {
    case 'SET_ROW_IDENTIFIERS_AND_STANDARD_FIELDS':
      return {
        importFileSetups: state.importFileSetups,
        usersCurrentPage: 'main',
        confirmRowIdentifiers: false,
        importConfirmed: false,
        importFileNameConfirmed: false,
        importRowIdentifierValues: [],
        importSetupArray: [], // holds import setup values to be matched with formMappingArray from DB
        formMappingArray: [], // holds a dump of that form's mappings from DB
        importDataArray: [], // holds the final cut of data to be imported to the form
        standardFields: action.standardFields,
        rowIdentifiers: action.rowIdentifiers,
        importRowIdentifierStandardFieldName: 'Address',
        mapImportToStandardFieldsERROR: false,
        promptUserOnStartOver: false,
        selectedImportFileSetupId: null,
        selectedImportFileSetupText: null,
        selectedFormId: undefined,
        selectedFormName: null,
        selectedFormDescription: null,
        selectedFormPublic: false,
        selectedFormConfirmedForImport: false,
        formConfirmed: false,
        importFieldMappingDbUpdates: [],
        preMapConfirmed: false,
        userIsMappingForm: false,
        formFieldSelector: null,
        formMappingStandardFieldId: null,
        formMappingRowIdentifierId: null,
        enteredDefaultValue: null,
        defaultValueConfirmed: false,
        enteredDefaultOverride: null,
        defaultOverrideConfirmed: false,
        rightClickedFormElementType: null,
        rightClickedFormElementValue: null,
        rightClickedFormElementOptions: null,
        rightClickSelectionIsValid: false,
        totalFieldsPopulated: null, 
        importFieldsPopulated: null,
        defaultFieldsPopulated: null,
        fillFormFailed: false
      }
    case 'STORE_IMPORT_FILE_SETUPS':
      return {
        ...state,
        importFileSetups: action.importFileSetups
      }
    case 'STORE_RAW_IMPORT_DATA':
      return {
        ...state,
        importedData: action.importedData,
        standardFields: action.standardFields,
        importedFieldNames: action.importedFieldNames,
        importRowIdentifierField: action.importRowIdentifierField,
        importRowIdentifierValues: action.importRowIdentifierValues
      }
    case 'UPDATE_IMPORT_FILE_MAPPING_IN_STORE':
      return {
        ...state,
        standardFields: action.standardFields,
        importRowIdentifierField: action.importRowIdentifierField
      }
    case 'UPDATE_IMPORT_FILE_MAPPING_IN_STORE_AND_FLAG_FOR_DB_UPDATE':
      return {
        ...state,
        standardFields: action.standardFields,
        importRowIdentifierField: action.importRowIdentifierField,
        importFieldMappingDbUpdates: action.newImportFieldMappingDbUpdates
      }
    case 'COMPLETE_UPDATE_IMPORT_FIELD_MAPPINGS_IN_DB':
      return {
        ...state,
        importFieldMappingDbUpdates: action.importFieldMappingDbUpdates
      }
    case 'SET_MAP_IMPORT_TO_STANDARDFIELDS_ERROR':
      return {
        ...state,
        mapImportToStandardFieldsERROR: true
      }
    case 'CONFIRM_IMPORT':
      return {
        ...state,
        importConfirmed: action.importConfirmed,
        standardFields: action.standardFields,
        mapImportToStandardFieldsERROR: false
      }
    case 'NAME_IMPORT_FILE':
      return {
        ...state,
        selectedImportFileSetupName: action.selectedImportFileSetupName
      }
    case 'COMPLETE_CONFIRM_NEW_IMPORT_FILE_NAME':
      return {
        ...state,
        importFileNameConfirmed: action.importFileNameConfirmed,
        selectedImportFileSetupId: action.selectedImportFileSetupId
      }
    case 'CONFIRM_EXISTING_IMPORT_FILE_NAME':
      return {
        ...state,
        importFileNameConfirmed: action.importFileNameConfirmed
      }
    case 'SET_IMPORT_ROW_IDENTIFIER_VALUE':
      return {
        ...state,
        rowIdentifiers: action.rowIdentifiers
      }
    case 'CONFIRM_ROW_IDENTIFIERS':
      return {
        ...state,
        confirmRowIdentifiers: action.confirmRowIdentifiers,
        importSetupArray: action.importSetupArray
      }
    case 'SET_SELECTED_IMPORT_FILE_SETUP':
      return {
        ...state,
        selectedImportFileSetupId: action.selectedImportFileSetupId,
        selectedImportFileSetupName: action.selectedImportFileSetupName
      }
    case 'STORE_IMPORT_FIELD_MAPPINGS':
      return {
        ...state,
        standardFields: action.standardFields
      }
    case 'TOGGLE_PROMPT_USER_ON_START_OVER':
      return {
        ...state,
        promptUserOnStartOver: action.promptUserOnStartOver
      }
    case 'UPDATE_USERS_CURRENT_PAGE':
      return {
        ...state,
        usersCurrentPage: action.usersCurrentPage
      }
    case 'SET_SELECTED_FORM':
      return {
        ...state,
        selectedFormId: action.selectedFormId,
        selectedFormName: action.selectedFormName,
        selectedFormDescription: action.selectedFormDescription,
        selectedFormPublic: action.selectedFormPublic,
        selectedFormConfirmedForImport: action.selectedFormConfirmedForImport
      }
    case 'NAME_FORM':
      return {
        ...state,
        selectedFormName: action.selectedFormName
      }
    case 'DESCRIBE_FORM':
      return {
        ...state,
        selectedFormDescription: action.selectedFormDescription
      }
    case 'SET_SELECTED_FORM_ID':
      return {
        ...state,
        selectedFormId: action.selectedFormId,
        formConfirmed: action.formConfirmed
      }
    case 'STORE_USER_FORMS':
      return {
        ...state,
        userForms: action.userForms
      }
    case 'CONFIRM_EXISTING_FORM':
      return {
        ...state,
        formConfirmed: action.formConfirmed
      }
    case 'CONFIRM_UPDATE_FORM':
      return {
        ...state,
        formConfirmed: action.formConfirmed
      }
    case 'CONFIRM_PRE_MAP':
      return {
        ...state,
        preMapConfirmed: action.preMapConfirmed
      }
    case 'SET_USER_IS_MAPPING_FORM':
      return {
        ...state,
        userIsMappingForm: action.userIsMappingForm,
        confirmRowIdentifiers: action.confirmRowIdentifiers
      }
    case 'SET_SELECTED_FORM_PUBLIC':
      return {
        ...state,
        selectedFormPublic: action.selectedFormPublic
      }
    case 'SET_FORM_FIELD_SELECTOR':
      return {
        ...state,
        formFieldSelector: action.formFieldSelector
      }
    case 'SET_FORM_MAPPING_DATA':
      return {
        ...state,
        formMappingRowIdentifierId: action.formMappingRowIdentifierId,
        formMappingStandardFieldId: action.formMappingStandardFieldId
      }
    case 'RESET_FORM_MAPPING_FIELDS':
      return {
        ...state,
        formMappingRowIdentifierId: action.formMappingRowIdentifierId,
        formMappingStandardFieldId: action.formMappingStandardFieldId,
        enteredDefaultValue: action.enteredDefaultValue,
        enteredDefaultOverride: action.enteredDefaultOverride,
        rightClickedFormElementType: action.rightClickedFormElementType,
        rightClickedFormElementValue: action.rightClickedFormElementValue,
        rightClickedFormElementOptions: action.rightClickedFormElementOptions,
        defaultValueConfirmed: action.defaultValueConfirmed,
        defaultOverrideConfirmed: action.defaultOverrideConfirmed,
        formFieldSelector: action.formFieldSelector,
        rightClickSelectionIsValid: action.rightClickSelectionIsValid,
        totalFieldsPopulated: action.totalFieldsPopulated, 
        importFieldsPopulated: action.importFieldsPopulated,
        defaultFieldsPopulated: action.defaultFieldsPopulated
      }
      case 'STORE_FORM_MAPPINGS':
      return {
        ...state,
        formMappingArray: action.formMappingArray,
        importDataArray: action.importDataArray
      }
      case 'SET_DEFAULT_VALUE':
      return {
        ...state,
        enteredDefaultValue: action.enteredDefaultValue
      }
      case 'SET_DEFAULT_OVERRIDE':
      return {
        ...state,
        enteredDefaultOverride: action.enteredDefaultOverride
      }
      case 'TOGGLE_CONFIRM_DEFAULT_VALUE':
      return {
        ...state,
        defaultValueConfirmed: action.defaultValueConfirmed
      }
      case 'TOGGLE_CONFIRM_DEFAULT_OVERRIDE':
      return {
        ...state,
        defaultOverrideConfirmed: action.defaultOverrideConfirmed
      }
      case 'SET_RIGHT_CLICKED_FORM_ELEMENT_ITEMS':
      return {
        ...state,
        rightClickedFormElementType: action.rightClickedFormElementType,
        rightClickedFormElementValue: action.rightClickedFormElementValue,
        rightClickedFormElementOptions: action.rightClickedFormElementOptions
      }
      case 'STORE_FORM_MAPPINGS_DIRECT_FROM_DB':
      return {
        ...state,
        formMappingArray: action.formMappingArray
      }
      case 'SET_DEFAULT_MODAL_ITEMS':
      return {
        ...state,
        enteredDefaultValue: action.enteredDefaultValue,
        enteredDefaultOverride: action.enteredDefaultOverride
      }
      case 'SET_RIGHT_CLICK_SELECTION_IS_VALID':
      return {
        ...state,
        rightClickSelectionIsValid: action.rightClickSelectionIsValid
      }
      case 'SET_POST_FILL_FORM_FIELDS':
      return {
        ...state,
        totalFieldsPopulated: action.totalFieldsPopulated, 
        importFieldsPopulated: action.importFieldsPopulated,
        defaultFieldsPopulated: action.defaultFieldsPopulated
      }
      case 'CLEAR_FILL_FORM_RESULTS':
      return {
        ...state,
        totalFieldsPopulated: action.totalFieldsPopulated,
        importFieldsPopulated: action.importFieldsPopulated,
        defaultFieldsPopulated: action.defaultFieldsPopulated, 
        confirmRowIdentifiers: action.confirmRowIdentifiers,
        selectedFormConfirmedForImport: action.selectedFormConfirmedForImport
      }
      case 'SET_FILL_FORM_FAILED':
      return {
        ...state,
        fillFormFailed: action.fillFormFailed
      }
    default:
      return state
  }
}