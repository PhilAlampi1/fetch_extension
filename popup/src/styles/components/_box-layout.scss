.box-layout {
    background: url('/images/buildings.jpeg');
    background-size: cover;
    display: flex;
    height: 100vh;
    width: 100vw;
}

.box-layout__box {
    background: fade-out(black, 0.25);
    border-radius: 3px;
    text-align: center;
    width: 25rem;
    margin: auto;
}

.box-layout__title {
    color: white;
    font-size: $xl-size;
    font-weight: bold;
}

.box-layout__subtext {
    color: white;
    font-size: $m-size;
    font-weight: 500;
    padding: 0 0 $s-size 0;
    text-align: center;
}

.box-layout__footer {
    display: flex;
    justify-content: space-around;
}

.box-layout__footertext {
    color: white;
    font-size: $s-size;
    margin: $s-size;
}

.box-layout__disclaimer {
    margin: $xl-size 0 0 0;
}