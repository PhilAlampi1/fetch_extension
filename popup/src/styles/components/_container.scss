.container {
    margin: 0 auto;
    max-width: 60rem;
    padding: 0 $m-size;
    overflow: scroll;
}

.container__centertext {
    text-align: center;
    margin: $xl-size 0 0 0;
    overflow: scroll;
}

.container__lefttext {
    text-align: left;
    margin: 0;
    overflow: scroll;
}

.container__flex {
    display: flex;
    justify-content: space-around;
    overflow: scroll;
}

.container--center {
    display: flex;
    align-content: center;
    justify-content: center;
    overflow: scroll;
}

.container__flexvertical {
    display: flex;
    flex-direction: column;
  }

.container__addspace {
    margin: $s-size 0;
}

.container__centerify {
    display: flex;
    align-items: center;
    text-align: center;
    margin: $xl-size 0 0 0;
    overflow: scroll;
}

.container__child {
    flex: 1;
    overflow: scroll;
}

.add-space {
    margin: $xl-size 0 0 0;
}