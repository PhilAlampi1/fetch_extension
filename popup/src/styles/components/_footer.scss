.footer {
    display: flex;
    justify-content: space-around;
}

.footer-item {
    color: black;
    font-size: $m-size;
    margin: $xl-size $s-size 1rem $s-size;
}