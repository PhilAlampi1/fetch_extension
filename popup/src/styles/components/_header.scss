.header {
    align-items: flex-start;
    background: $off-black;
    color: white;
    display: flex;
    padding: $m-size 0;
}

.header__container {
    align-items: center;
    display: flex;
    flex: 1;
    justify-content: space-between;
    padding: 0 $m-size;
}

.header__title {
    font-size: $l-size;
    margin: 0 $m-size;
    padding: $s-size;
    text-align: left;
    flex: 1;
}

.header__linkgroup {
    flex: 4;
}

.header__links {
    color: $off-white;
    font-size: $m-size;
    font-weight: 500;
    margin: $s-size;
    padding: $s-size;
    text-align: center;
}

.header__span {
    width: 100%
}

.header-question-parent {
    align-items: center;
    display: flex;
    justify-content: flex-start;
}

.header-question-child {
    font-size: $m-size;
    margin: 0 $s-size;
    padding: $s-size;
    width: 100px;
}