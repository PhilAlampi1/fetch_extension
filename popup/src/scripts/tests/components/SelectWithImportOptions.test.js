import React from 'react'
import { shallow } from 'enzyme'
import { SelectWithImportOptions } from '../../components/SelectWithImportOptions'
import { importFieldOptions } from '../../../../../event/src/tests/fixtures/imports'
// import auth from '../../../../../event/src/tests/fixtures/auth'

// test('', () => {

// })

const onChangeSpy = jest.fn()
const importedFieldName = "Number of Bathrooms"
const localImportFieldOptions = importFieldOptions
const standardFieldId = "4"
test('SelectWithImportOptions renders correctly with inputs', () => {
    const wrapper = shallow(<SelectWithImportOptions
        handleOnChange={onChangeSpy}
        importFieldOptions={localImportFieldOptions}
        importedFieldName
        standardFieldId
    />)
    expect(wrapper).toMatchSnapshot()
})

test('SelectWithImportOptions select element calls handleOnChange with correct values on change', () => {
    const e = {
        target: {
            value: "Property Type"
        }   
    }
    const wrapper = shallow(<SelectWithImportOptions
        handleOnChange={onChangeSpy}
        importFieldOptions={localImportFieldOptions}
        importedFieldName
        standardFieldId
    />)
    wrapper.find('select').simulate('change', e)
    expect(onChangeSpy).toHaveBeenCalled()
})