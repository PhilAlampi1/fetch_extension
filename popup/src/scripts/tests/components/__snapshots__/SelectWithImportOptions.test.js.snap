// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectWithImportOptions renders correctly with inputs 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectWithImportOptions
    handleOnChange={[Function]}
    importFieldOptions={
      Array [
        "Street Address",
        "Property Type",
        "IDX Address Mapg",
      ]
    }
    importedFieldName={true}
    standardFieldId={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <select
        onChange={[Function]}
        value={true}
      >
        <option
          value=""
        >
          No mapping
        </option>
        <option
          value="Street Address"
        >
          Street Address
        </option>
        <option
          value="Property Type"
        >
          Property Type
        </option>
        <option
          value="IDX Address Mapg"
        >
          IDX Address Mapg
        </option>
      </select>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value=""
          >
            No mapping
          </option>,
          Array [
            <option
              value="Street Address"
            >
              Street Address
            </option>,
            <option
              value="Property Type"
            >
              Property Type
            </option>,
            <option
              value="IDX Address Mapg"
            >
              IDX Address Mapg
            </option>,
          ],
        ],
        "onChange": [Function],
        "value": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "No mapping",
            "value": "",
          },
          "ref": null,
          "rendered": "No mapping",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Street Address",
            "value": "Street Address",
          },
          "ref": null,
          "rendered": "Street Address",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Property Type",
            "value": "Property Type",
          },
          "ref": null,
          "rendered": "Property Type",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "IDX Address Mapg",
            "value": "IDX Address Mapg",
          },
          "ref": null,
          "rendered": "IDX Address Mapg",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <select
          onChange={[Function]}
          value={true}
        >
          <option
            value=""
          >
            No mapping
          </option>
          <option
            value="Street Address"
          >
            Street Address
          </option>
          <option
            value="Property Type"
          >
            Property Type
          </option>
          <option
            value="IDX Address Mapg"
          >
            IDX Address Mapg
          </option>
        </select>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value=""
            >
              No mapping
            </option>,
            Array [
              <option
                value="Street Address"
              >
                Street Address
              </option>,
              <option
                value="Property Type"
              >
                Property Type
              </option>,
              <option
                value="IDX Address Mapg"
              >
                IDX Address Mapg
              </option>,
            ],
          ],
          "onChange": [Function],
          "value": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "No mapping",
              "value": "",
            },
            "ref": null,
            "rendered": "No mapping",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "Street Address",
              "value": "Street Address",
            },
            "ref": null,
            "rendered": "Street Address",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "Property Type",
              "value": "Property Type",
            },
            "ref": null,
            "rendered": "Property Type",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "IDX Address Mapg",
              "value": "IDX Address Mapg",
            },
            "ref": null,
            "rendered": "IDX Address Mapg",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
